sdks/java/core/src/main/java/org/apache/beam/sdk/util/ApiSurface.java

```diff
@@ -86,7 +86,7 @@
  * not interesting.
  */
 @SuppressWarnings("rawtypes")
-public class ApiSurface {
+public abstract  class ApiSurface {
```

Comment:
I don't think we need to make this abstract and have subclasses extend `ApiSurface`. Really, we just need to build values of type `ApiSurface`. Something like this: `directRunnerApiSurface() { return <build the API surface>; }`.